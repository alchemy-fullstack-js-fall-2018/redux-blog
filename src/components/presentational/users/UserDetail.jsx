import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import PostList from '../posts/PostList.jsx';

export default class UserDetail extends PureComponent {

  static propTypes = {
    user: PropTypes.object.isRequired,
    posts: PropTypes.array.isRequired
  };

  render() {
    const {
      id, name, username, email,
      address, phone, website, company
    } = this.props.user;
    const { posts } = this.props;

    // <p>address: {address}</p>
    // <p>company: {company}</p>

    return (
      <div>
        <p>id: {id}</p>
        <p>name: {name}</p>
        <p>username: {username}</p>
        <p>email: {email}</p>

        <p>phone: {phone}</p>
        <p>website: {website}</p>
        <PostList posts={posts} />
      </div>
    );

  }
}
