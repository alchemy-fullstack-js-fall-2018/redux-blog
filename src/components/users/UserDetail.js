import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

export default class UserDetail extends Component {
  static propTypes = {
    user: PropTypes.object,
    fetchUser: PropTypes.func,
    match: PropTypes.object
  };

  componentDidMount() {
    const { fetchUser, match } = this.props;
    fetchUser(match.params.id);
  }

  render() {
    const { user } = this.props;
    if(!user) return null;
    return (
      <Fragment>
        <h2>User Detail</h2>
        <h3>Username: {user.username}</h3>
        <p>Name: {user.name}</p>
        <p>Email: {user.email}</p>
      </Fragment>
    );
  }
}
