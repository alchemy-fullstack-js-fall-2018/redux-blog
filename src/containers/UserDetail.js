import { connect } from 'react-redux';
import UserDetail from '../components/users/UserDetail';
import { getUsers } from '../selectors/users';
import { fetchUser } from '../actions/users';
import { loadingWithParagraph } from '../components/loading/LoadingFallback';

const mapStateToProps = state => ({
  user: getUsers(state)[0] || {}
});

const mapDispatchToProps = dispatch => ({
  fetchUser(id) {
    dispatch(fetchUser(id));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(loadingWithParagraph(UserDetail));
