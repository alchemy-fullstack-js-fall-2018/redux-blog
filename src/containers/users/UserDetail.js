import { connect } from 'react-redux';
import UserDetail from '../../components/users/UserDetail';
import { fetchUser } from '../../actions/users';
import { getUser } from '../../selectors/users';

const mapStateToProps = (state, props) => ({
  user: getUser(state, props.match.params.id),
});

const mapDispatchToProps = dispatch => ({
  fetchUser(id) {
    dispatch(fetchUser(id));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UserDetail);
