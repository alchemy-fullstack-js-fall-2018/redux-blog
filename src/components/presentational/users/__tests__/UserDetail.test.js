import React from 'react';
import { shallow, mount } from 'enzyme';
import UserDetail from '../UserDetail.jsx';
import mockUsers from '../../../../testing/fixtures/users.json';
import mockPosts from '../../../../testing/fixtures/posts.json';

describe('<UserDetail />', () => {
  test('renders', () => {
    const user = mockUsers[0];
    const posts = mockPosts.filter(post => post.userId === 1);
    const wrapper = shallow(
      <UserDetail
        user={user}
        posts={posts}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
