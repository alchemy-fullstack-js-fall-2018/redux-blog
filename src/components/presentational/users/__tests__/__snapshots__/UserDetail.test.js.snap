// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetail /> renders 1`] = `
<div>
  <p>
    id: 
    1
  </p>
  <p>
    name: 
    Leanne Graham
  </p>
  <p>
    username: 
    Bret
  </p>
  <p>
    email: 
    Sincere@april.biz
  </p>
  <p>
    phone: 
    1-770-736-8031 x56442
  </p>
  <p>
    website: 
    hildegard.org
  </p>
  <PostList
    posts={
      Array [
        Object {
          "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
          "id": 1,
          "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
          "userId": 1,
        },
        Object {
          "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
          "id": 2,
          "title": "qui est esse",
          "userId": 1,
        },
        Object {
          "body": "et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut",
          "id": 3,
          "title": "ea molestias quasi exercitationem repellat qui ipsa sit aut",
          "userId": 1,
        },
        Object {
          "body": "ullam et saepe reiciendis voluptatem adipisci
sit amet autem assumenda provident rerum culpa
quis hic commodi nesciunt rem tenetur doloremque ipsam iure
quis sunt voluptatem rerum illo velit",
          "id": 4,
          "title": "eum et est occaecati",
          "userId": 1,
        },
        Object {
          "body": "repudiandae veniam quaerat sunt sed
alias aut fugiat sit autem sed est
voluptatem omnis possimus esse voluptatibus quis
est aut tenetur dolor neque",
          "id": 5,
          "title": "nesciunt quas odio",
          "userId": 1,
        },
        Object {
          "body": "ut aspernatur corporis harum nihil quis provident sequi
mollitia nobis aliquid molestiae
perspiciatis et ea nemo ab reprehenderit accusantium quas
voluptate dolores velit et doloremque molestiae",
          "id": 6,
          "title": "dolorem eum magni eos aperiam quia",
          "userId": 1,
        },
        Object {
          "body": "dolore placeat quibusdam ea quo vitae
magni quis enim qui quis quo nemo aut saepe
quidem repellat excepturi ut quia
sunt ut sequi eos ea sed quas",
          "id": 7,
          "title": "magnam facilis autem",
          "userId": 1,
        },
        Object {
          "body": "dignissimos aperiam dolorem qui eum
facilis quibusdam animi sint suscipit qui sint possimus cum
quaerat magni maiores excepturi
ipsam ut commodi dolor voluptatum modi aut vitae",
          "id": 8,
          "title": "dolorem dolore est ipsam",
          "userId": 1,
        },
        Object {
          "body": "consectetur animi nesciunt iure dolore
enim quia ad
veniam autem ut quam aut nobis
et est aut quod aut provident voluptas autem voluptas",
          "id": 9,
          "title": "nesciunt iure omnis dolorem tempora et accusantium",
          "userId": 1,
        },
        Object {
          "body": "quo et expedita modi cum officia vel magni
doloribus qui repudiandae
vero nisi sit
quos veniam quod sed accusamus veritatis error",
          "id": 10,
          "title": "optio molestias id quia eum",
          "userId": 1,
        },
      ]
    }
  />
</div>
`;
